File: online_review_deliverables/trunk/src/java/main/com/topcoder/management/deliverable/persistence/PersistenceException.java
/*
 * Copyright (c) 2006, TopCoder, Inc. All rights reserved.
 */
package com.topcoder.management.deliverable.persistence;

import com.topcoder.util.errorhandling.BaseException;

/**
 * @author kr00tki
 * @version 1.0
 */
public class PersistenceException extends BaseException {
    /**
     * DeliverableCheckingException constructor: Creates a new DeliverableCheckingException .
     *
     *
     * @param message Explanation of error, can be null
     */
    public PersistenceException(String message) {
        super(message);
    }

    /**
     * DeliverableCheckingException constructor: Creates a new DeliverableCheckingException .
     *
     *
     * @param message Explanation of error, can be null
     * @param cause Underlying cause of error, can be null
     */
    public PersistenceException(String message, Exception cause) {
        super(message, cause);
    }
}

File: client_logic_for_msie/trunk/src/csharp/main/Orpheus/Plugin/InternetExplorer/PersistenceException.cs
/*
 * Copyright (C) 2006 TopCoder Inc., All Rights Reserved.
 *
 * PersistenceException.cs
 */

using System;
using System.Runtime.Serialization;

namespace Orpheus.Plugin.InternetExplorer
{
    /// <summary>
    /// Exception to signal any problems with the persistence.
    /// </summary>
    ///
    /// <author>TCSDESIGNER</author>
    /// <author>TCSDEVELOPER</author>
    /// <version>1.0</version>
    /// <copyright>Copyright (C) 2006 TopCoder Inc., All Rights Reserved.</copyright>
    [Serializable]
    public class PersistenceException : ClientLogicExtensionException
    {
        /// <summary>
        /// Creates a new exception.
        /// </summary>
        public PersistenceException()
            : base()
        {
        }

        /// <summary>
        /// Creates a new exception with the specified message.
        /// </summary>
        ///
        /// <param name="message">Exception message.</param>
        public PersistenceException(string message)
            : base(message)
        {
        }

        /// <summary>
        /// Creates a new exception with the specified message and cause
        /// </summary>
        ///
        /// <param name="message">Exception message.</param>
        /// <param name="cause">Exception cause.</param>
        public PersistenceException(string message, Exception cause)
            : base(message, cause)
        {
        }

        /// <summary>
        /// Seriaization constructor.
        /// </summary>
        ///
        /// <param name="info">Serialization info.</param>
        /// <param name="context">Streaming context</param>
        protected PersistenceException(SerializationInfo info, StreamingContext context)
            : base(info, context)
        {
        }
    }
}
